### PR TITLE
Extend tutorial to include publishing

### LINKED ISSUE #12735: Self Contained Executable
We could use an example on running one of these examples as a self contained console app on linux.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9fa0c79e-1729-15f7-52e4-0c9afff43ec9
* Version Independent ID: 1c3c5d30-9690-8c4a-4e2c-2ce5f4386898
* Content: [Get started with .NET Core using the CLI - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/tutorials/using-with-xplat-cli#feedback)
* Content Source: [docs/core/tutorials/using-with-xplat-cli.md](https://github.com/dotnet/docs/blob/master/docs/core/tutorials/using-with-xplat-cli.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @cartermp
* Microsoft Alias: **mairaw**

### PR DESCRIPTION
**Pull request by Thraka**
## Summary

This extends the .NET Core CLI 2.1 tutorial to include a publishing section at the end. I'll file a new issue to track updating this to 3.0 which includes an additional publishing method.

Fixes #12735

### CODE REVIEW

**Review by mairaw — CHANGES_REQUESTED** [review]
I think it should be easy to upgrade this article to 3.0 in this pass as well and avoid another round of reviews. 
I'm also on the fence on whether to include these publishing steps here on this tutorial. Up until the publishing steps that were added, the tutorial works for all platforms. Then, it enters into a mix of conceptual material + generic steps. Adding @KathleenDollard and @rpetrusha for a second opinion.

File: docs/core/tutorials/using-with-xplat-cli.md
Comment (Thraka) [reply docs/core/tutorials/using-with-xplat-cli.md]: Can we get loc to change? `\` is a markdown tag, they should be adapting to it since its the writing system we use.

Also, the `01. ` is amazing and perfect because it creates your alignment for child content at 4-spaces instead of 3. :)

**Comment by Thraka** [comment]
OK after thinking and reading this article over and over I think it's best to just simplify the publishing part as a step in the tutorial. I got rid of the conceptual stuff. It didn't flow well.

File: docs/core/tutorials/using-with-xplat-cli.md
Comment (mairaw) [inline docs/core/tutorials/using-with-xplat-cli.md]: publish folder?

```suggestion
Once you're ready to distribute your app, use the [`dotnet publish`](../tools/dotnet-publish.md) command to generate the _publish_ folder at _bin\\debug\\netcoreapp2.1\\publish\\_ (use `/` for non-Windows systems). You can distribute the contents of the _publish_ folder to other platforms as long as they've already installed the dotnet runtime.
```

File: docs/core/tutorials/using-with-xplat-cli.md
Comment (mairaw) [inline docs/core/tutorials/using-with-xplat-cli.md]:
```suggestion
You can run your published app with the [dotnet](../tools/dotnet.md) command:
```

File: docs/core/tutorials/using-with-xplat-cli.md
Comment (mairaw) [inline docs/core/tutorials/using-with-xplat-cli.md]: you should update the date in this case. It's a significant change.

File: docs/core/tutorials/using-with-xplat-cli.md
Comment (mairaw) [inline docs/core/tutorials/using-with-xplat-cli.md]: This no longer makes sense given what you've just added but some of the information on this paragraph should be moved to the previous section.

**Review by mairaw — APPROVED** [review]
I think it looks better now. I'm approving this to unblock you but I left some comments for you consider before merging.

**Pull request merged**
